src/components/homeScreen/RandomMeals.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from "react";
+import {
+  View,
+  FlatList,
+  Text,
+  Image,
+  StyleSheet,
+  ActivityIndicator,
+  TouchableOpacity,
+} from "react-native";
+import { FetchRandomMealResponse, Meal } from "../../types/types";
+import businessManagerService from "../../services";
+import { dimensions } from "../../styles/branding";
+import { primary } from "../../styles/styleGuide";
+import { useIsFocused } from "@react-navigation/native";
+import Constants from "../../utils/constants";
+import { generalStyles } from "../../styles/generalStyleSheet";
+
+const RandomMealsList = () => {
+  const [ramdomMeals, onChangeRandomMeals] = useState<Set<Meal>>(new Set());
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const isFocused = useIsFocused();
+
+  const renderItem = ({ item }: { item: Meal }) => (
+    <TouchableOpacity onPress={onClick}>
+      <View style={styles.itemStyle}>
+        <Image
+          source={{ uri: item.strMealThumb }}
+          style={styles.imageStyle}
+          resizeMode="cover"
+        ></Image>
+        <View style={styles.textViewStyle}>
+          <Text style={[generalStyles.fontStyle, styles.textStyle]}>
+            {item.strMeal}
+          </Text>
+        </View>
+      </View>
+    </TouchableOpacity>
+  );
+
+  const getRandomMeal = async () => {
+    try {
+      setLoading(true);
+      const response: FetchRandomMealResponse =
+        await businessManagerService.fetchRandomMealApi();
+      response?.meals.map((meal) => {
+        onChangeRandomMeals((prevSet) => {
+          const newSet = new Set(prevSet);
+          newSet.add(meal);
+          return newSet;
+        });
+      });
+    } catch (e) {
+      console.error("[ CATEGORIES]", e);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    onChangeRandomMeals(new Set());
+    for (let i = 1; i <= Constants.RANDOM_MEALS_SIZE; i++) {
+      getRandomMeal();
+    }
+  }, [isFocused]);
+
+  return (
+    <View>
+      {isLoading ? (
+        <View style={{ flex: 1, padding: 200 }}>
+          <ActivityIndicator
+            size={dimensions.loadingSize}
+            color={primary.green}
+          />
+        </View>
+      ) : (
+        <View>
+          <FlatList<Meal>
+            data={Array.from(ramdomMeals)}
```

Comment:
why Array.from?

Reply:
I used a set instead of an array as we don't need duplicated meals while we generate them and therefore I needed to use Array.from(set) because flatlist works only with arrays